### PR TITLE
set sensible cache-control headers on static assets

### PR DESCRIPTION
**Pull request by loumoore**
a first step towards better front end performance. (next up, CDN!)

### CODE REVIEW

**Comment by loumoore** [comment]
for acceptance do: curl -I https://demo.getyourrefund.org/assets/some-asset-path and validate this header is present:
```
Cache-Control: public, max-age=31536000
```

File: config/environments/demo.rb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 require_relative "./shared_deployment_config"
 
 Rails.application.configure do
+  config.public_file_server.headers = {
+      'Cache-Control' => 'public, max-age=31536000'
+  }
```
Comment (bengolder) [inline config/environments/demo.rb:6]: If this belongs in both demo and prod, it might make more sense to put it in `shared_deployment_config.rb`

Comment (loumoore) [reply config/environments/demo.rb:6]: ah my bad for some reason i thought that included staging but i see that's separate. i'll do that.

File: config/environments/shared_deployment_config.rb [hunk 2]
```diff
@@ -18,9 +18,12 @@
   # or in config/master.key. This key is used to decrypt credentials (and other encrypted files).
   # config.require_master_key = true
 
-  # Disable serving static files from the `/public` folder by default since
-  # Apache or NGINX already handles this.
+  # Serve static files from the `/public` folder based on environment settings
   config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
+  # Set cache control headers
+  config.public_file_server.headers = {
```
Comment (alaroia-cfa) [inline config/environments/shared_deployment_config.rb:24]: For anyone curious, `rails assets:precompile` will precompile assets on one's own machine

**Review by alaroia-cfa — APPROVED** [review]
This is great! Thanks @loumoore 

For testing on demo, once this auto-deploys to demo, one can use:

```
curl -I https://demo.getyourrefund.org/assets/vita-logo-9281a2c4c6c0191e8432aa44532fa842083db858dc730d57bae6a3cfba1c45b9.png
```

You can expect to see `Cache-Control: public, max-age=31536000` in the output. (it's case insensitive, so it's fine if it's lowercase.)

**Pull request merged**